quickcomm/models.py

```diff
@@ -18,6 +24,145 @@
 # We used UUIDs for pkeys to be more secure.
 # TODO what are the constraints on fields being null?
 
+class HostAuthenticator(models.Model):
+    """A host authenticator is a username and password that can be used to
+    authenticate to a host."""
+
+    # Note: passwords are stored in plain-text. They are API keys, so they are
+    # not sensitive. If we want to store them in a more secure way, we can
+    # hash them, but for now, we will leave them as is.
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    username = models.CharField(max_length=100, help_text="The username to use to authenticate to the host.", verbose_name="Username")
+    password = models.CharField(max_length=100, help_text="The password to use to authenticate to the host.", verbose_name="Password")
+    nickname = models.CharField(max_length=100, help_text="The nickname of the host authenticator. This is only used for display purposes.", verbose_name="Nickname", null=True, blank=True)
```

Comment:
curious why the authenticator needs the nickname too? I would think it just needs username and password since the Host itself has the nickname